xdsl/dialects/riscv_scf.py

```diff
@@ -73,6 +74,41 @@ def __init__(
             regions=[body],
         )
 
+    def verify_(self):
+        if (len(self.iter_args) + 1) != len(self.body.block.args):
+            raise VerifyException(
+                f"Wrong number of block arguments, expected {len(self.iter_args)+1}, got "
+                f"{len(self.body.block.args)}. The body must have the induction "
+                f"variable and loop-carried variables as arguments."
+            )
+        if self.body.block.args and (iter_var := self.body.block.args[0]):
+            if not isinstance(iter_var.type, IntRegisterType):
+                raise VerifyException(
+                    f"The first block argument of the body is of type {iter_var.type}"
+                    " instead of riscv.IntRegisterType"
+                )
+        for idx, arg in enumerate(self.iter_args):
+            if self.body.block.args[idx + 1].type != arg.type:
```

Comment:
Can we use `zip` instead of `enumerate` + `idx`?

Reply:
I like the idx version more for exceptions, but it's a small difference anyway. I'm not sure it's worth the time changing it one way or another.